### ISSUE TITLE
Dev Containers fail to create with Podman

### ISSUE THREAD

**Issue by ljcoogan** [microsoft/vscode-remote-release#10591]
- VSCode Version: 1.9.6.2
- Local OS Version: Ubuntu 24.04
- Remote OS Version: Alpine Linux (Dev Containers bootstrap image)
- Remote Extension/Connection Type: Dev Containers
- Logs: 
```
[31 ms] Dev Containers 0.394.0 in VS Code 1.96.2 (fabdb6a30b49f79a7aba0f2ad9df9b399473380f).
[30 ms] Start: Resolving Remote
[55 ms] Start: Check Docker is running
[55 ms] Start: Run: podman version
[100 ms] Client:       Podman Engine
Version:      4.9.3
API Version:  4.9.3
Go Version:   go1.22.2
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
[106 ms] Stop (51 ms): Run: podman version
[106 ms] Stop (51 ms): Check Docker is running
[111 ms] Start: Run: podman build -f /tmp/vsch-liam/bootstrap-image/0.394.0/bootstrap.Dockerfile -t vsc-volume-bootstrap /tmp/vsch-liam/bootstrap-image/0.394.0
STEP 1/9: FROM mcr.microsoft.com/devcontainers/base:0-alpine-3.20
STEP 2/9: COPY host-ca-certificates.crt /tmp/host-ca-certificates.crt
--> Using cache 57c092614d906477ebdb8d78731afb96ea72fe607f69d4d7099572ec562cddb2
--> 57c092614d90
STEP 3/9: RUN cat /tmp/host-ca-certificates.crt >> /etc/ssl/certs/ca-certificates.crt
--> Using cache 846828da3c0b09f6110a967bac2bda36a95d02d5745287ab84cabc860ef27143
--> 846828da3c0b
STEP 4/9: RUN csplit -f /usr/local/share/ca-certificates/host-ca-certificate- -b '%02d.pem' -z -s /tmp/host-ca-certificates.crt '/-----BEGIN CERTIFICATE-----/' '{*}'
--> Using cache f78e13b7b60428660eeb06cbc69b2d76c870b9a06bea7206db4719558441ae42
--> f78e13b7b604
STEP 5/9: ENV NODE_EXTRA_CA_CERTS=/etc/ssl/certs/ca-certificates.crt
--> Using cache 55b96e01cdf9462cb257bf4775ef91452ab7d7db84e6fe2dff236ee06e151c42
--> 55b96e01cdf9
STEP 6/9: RUN echo "@old https://dl-cdn.alpinelinux.org/alpine/v3.15/main" >> /etc/apk/repositories
--> Using cache 1fe8f93debb0b292400c8f40d8ccd648cd0ac6c0c5a1c450835dff76793192f0
--> 1fe8f93debb0
STEP 7/9: RUN apk add --no-cache        git-lfs         nodejs  python3         npm     make    g++     docker-cli      docker-cli-buildx      docker-cli-compose       openssh-client-default@old      ;
--> Using cache 20fb937906a8a6028b982cc23059f8b899662dc6e1ff2bdc656197e9d1d273d2
--> 20fb937906a8
STEP 8/9: RUN npm config set cafile /etc/ssl/certs/ca-certificates.crt && cd && npm i node-pty || echo "Continuing without node-pty."
--> Using cache 3075a61ae9566054add9052c58b5c4513f5729ef3856839d8456bbc4c0700208
--> 3075a61ae956
STEP 9/9: COPY .vscode-remote-containers /root/.vscode-remote-containers
--> Using cache bf48dcee1d9c494564bb08370071612f7708800746ddd4bb9cabc4e1b47148cd
COMMIT vsc-volume-bootstrap
--> bf48dcee1d9c
Successfully tagged localhost/vsc-volume-bootstrap:latest
bf48dcee1d9c494564bb08370071612f7708800746ddd4bb9cabc4e1b47148cd
[1180 ms] Stop (1069 ms): Run: podman build -f /tmp/vsch-liam/bootstrap-image/0.394.0/bootstrap.Dockerfile -t vsc-volume-bootstrap /tmp/vsch-liam/bootstrap-image/0.394.0
[1181 ms] Start: Run: podman run -d --mount type=volume,src=java,dst=/workspaces -v /var/run/docker.sock:/var/run/docker.sock vsc-volume-bootstrap sleep infinity
[1443 ms] Stop (262 ms): Run: podman run -d --mount type=volume,src=java,dst=/workspaces -v /var/run/docker.sock:/var/run/docker.sock vsc-volume-bootstrap sleep infinity
[1444 ms] Start: Run in container: /bin/sh
[1454 ms] Running Dev Containers CLI:   read-configuration --workspace-folder /workspaces/java --id-label vsc.devcontainer.volume.name=java --id-label vsc.devcontainer.volume.folder=java --log-level trace --log-format json --mount-workspace-git-root
[2080 ms] Start: Run in container: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.394.0/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /workspaces/java --id-label vsc.devcontainer.volume.name=java --id-label vsc.devcontainer.volume.folder=java --log-level trace --log-format json --mount-workspace-git-root
[2635 ms] @devcontainers/cli 0.72.0. Node.js v20.15.1. linux 6.8.0-51-generic x64.
[2635 ms] Start: Run: git rev-parse --show-cdup
[2649 ms] Stop (14 ms): Run: git rev-parse --show-cdup
[2651 ms] Start: Run: docker ps -q -a --filter label=vsc.devcontainer.volume.name=java --filter label=vsc.devcontainer.volume.folder=java
[2675 ms] Stop (24 ms): Run: docker ps -q -a --filter label=vsc.devcontainer.volume.name=java --filter label=vsc.devcontainer.volume.folder=java
[2676 ms] Command failed: docker ps -q -a --filter label=vsc.devcontainer.volume.name=java --filter label=vsc.devcontainer.volume.folder=java
[2734 ms] Stop (654 ms): Run in container: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.394.0/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /workspaces/java --id-label vsc.devcontainer.volume.name=java --id-label vsc.devcontainer.volume.folder=java --log-level trace --log-format json --mount-workspace-git-root
[2734 ms] Exit code 1
[2738 ms] Command failed: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.394.0/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /workspaces/java --id-label vsc.devcontainer.volume.name=java --id-label vsc.devcontainer.volume.folder=java --log-level trace --log-format json --mount-workspace-git-root
[2738 ms] Exit code 1
```

Steps to Reproduce:

1. Open new VS Code window with no project open
2. Set `"dev.containers.dockerPath": "podman",` in `settings.json`, as outlined [here](https://code.visualstudio.com/remote/advancedcontainers/docker-options#_podman).
3. Ctrl+Shift+P, select "Dev Containers: New Dev Container"
4. Select any dev container (Java, for example)
5. Select "Create Dev Container java"
6. Container will fail to create

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by ljcoogan** [comment]
I have just tested this in Fedora 41, which I thought might fix it as it uses a more recent release of Podman (5.3.1). Unfortunately, the same issue occurs. See log below:
```
[72 ms] Dev Containers 0.394.0 in VS Code 1.96.2 (fabdb6a30b49f79a7aba0f2ad9df9b399473380f).
[71 ms] Start: Resolving Remote
[123 ms] Start: Check Docker is running
[124 ms] Start: Run: podman version
[174 ms] Client:       Podman Engine
Version:      5.3.1
API Version:  5.3.1
Go Version:   go1.23.3
Built:        Thu Nov 21 00:00:00 2024
OS/Arch:      linux/amd64
[186 ms] Start: Run: podman build -f /tmp/vsch-liam/bootstrap-image/0.394.0/bootstrap.Dockerfile -t vsc-volume-bootstrap /tmp/vsch-liam/bootstrap-image/0.394.0
STEP 1/9: FROM mcr.microsoft.com/devcontainers/base:0-alpine-3.20
STEP 2/9: COPY host-ca-certificates.crt /tmp/host-ca-certificates.crt
--> Using cache 9c606d68a5f2809bc3fb5355477c86959db6e76b529bfcb0607bce9fd8b1d6c1
--> 9c606d68a5f2
STEP 3/9: RUN cat /tmp/host-ca-certificates.crt >> /etc/ssl/certs/ca-certificates.crt
--> Using cache 0a5efa197f1baa39dd02ac53155c8e666aa5e5ce8522f035bef6a9c4c68367da
--> 0a5efa197f1b
STEP 4/9: RUN csplit -f /usr/local/share/ca-certificates/host-ca-certificate- -b '%02d.pem' -z -s /tmp/host-ca-certificates.crt '/-----BEGIN CERTIFICATE-----/' '{*}'
--> Using cache a005a7fd6d7936f2c086b1127f73bc1d0649801a198105329862db87c3994416
--> a005a7fd6d79
STEP 5/9: ENV NODE_EXTRA_CA_CERTS=/etc/ssl/certs/ca-certificates.crt
--> Using cache 93dd21add31c08eba4bc59294d839aed61d9ff810f62fb03a6126895ff1ccf7e
--> 93dd21add31c
STEP 6/9: RUN echo "@old https://dl-cdn.alpinelinux.org/alpine/v3.15/main" >> /etc/apk/repositories
--> Using cache b07bbc676a99867b1bdac75b209e0dbca340865ecdfa8d2fc78723207d0455f1
--> b07bbc676a99
STEP 7/9: RUN apk add --no-cache        git-lfs         nodejs  python3         npm     make    g++     docker-cli      docker-cli-buildx       docker-cli-compose     openssh-client-default@old      ;
--> Using cache 45ae1cccce091fe1ecf98efac93d49a911e710daf6cf583c37d8dee2c32fb9f3
--> 45ae1cccce09
STEP 8/9: RUN npm config set cafile /etc/ssl/certs/ca-certificates.crt && cd && npm i node-pty || echo "Continuing without node-pty."
--> Using cache 93928f07807b31c387ae28fe82b0455c330871561e15e719552c1d819c08651b
--> 93928f07807b
STEP 9/9: COPY .vscode-remote-containers /root/.vscode-remote-containers
--> Using cache b18fca9f8be03110c4a7de9dedec39f38a08e2448a66b90b1a2139644f9bd343
COMMIT vsc-volume-bootstrap
--> b18fca9f8be0
Successfully tagged localhost/vsc-volume-bootstrap:latest
b18fca9f8be03110c4a7de9dedec39f38a08e2448a66b90b1a2139644f9bd343
[1554 ms] Start: Run: podman run -d --mount type=volume,src=node,dst=/workspaces -v /run/user/1000/podman/podman.sock:/var/run/docker.sock vsc-volume-bootstrap sleep infinity
[1794 ms] Start: Run in container: /bin/sh
[1804 ms] Running Dev Containers CLI:   read-configuration --workspace-folder /workspaces/node --id-label vsc.devcontainer.volume.name=node --id-label vsc.devcontainer.volume.folder=node --log-level debug --log-format json --mount-workspace-git-root
[2273 ms] Start: Run in container: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.394.0/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /workspaces/node --id-label vsc.devcontainer.volume.name=node --id-label vsc.devcontainer.volume.folder=node --log-level debug --log-format json --mount-workspace-git-root
[2783 ms] @devcontainers/cli 0.72.0. Node.js v20.15.1. linux 6.12.5-200.fc41.x86_64 x64.
[2783 ms] Start: Run: git rev-parse --show-cdup
[2799 ms] Start: Run: docker ps -q -a --filter label=vsc.devcontainer.volume.name=node --filter label=vsc.devcontainer.volume.folder=node
[2822 ms] Command failed: docker ps -q -a --filter label=vsc.devcontainer.volume.name=node --filter label=vsc.devcontainer.volume.folder=node
[2879 ms] Exit code 1
[2888 ms] Command failed: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.394.0/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /workspaces/node --id-label vsc.devcontainer.volume.name=node --id-label vsc.devcontainer.volume.folder=node --log-level debug --log-format json --mount-workspace-git-root
[2888 ms] Exit code 1
```

**Comment by ljcoogan** [comment]
I figured it out - I think it's an issue with Mandatory Access Control tools.

On my Fedora box, I have been able to get Dev Containers working by manually creating a docker-compose.yml, asking Dev Containers to create a dev container *from* that docker-compose.yml, and then appending the :Z flag to `../workspaces` in the generated docker-compose.yml file. I imagine there is a similar solution for AppArmor.

Closing bug.
